spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -71,11 +77,21 @@ protected TableCatalog tableCatalog() {
   }
 
   protected <T> T modifyIcebergTable(Identifier ident, Function<org.apache.iceberg.Table, T> func) {
-    return execute(ident, true, func);
+    try {
+      T result = execute(ident, true, func);
+      return result;
```

Comment:
Optional, can it be on oneline as before:
```
try {
    return execute(ident, true, func);
} finally {
```

Reply:
Good call. I previously had the close after the execute, but realized I wanted it in a finally block